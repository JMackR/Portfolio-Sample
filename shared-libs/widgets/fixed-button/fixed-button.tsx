import React from 'react'
import { Button, ButtonSize, ButtonType } from '@sample/basic-controls'
import { Margin } from '@sample/layout-controls'
import { StyleSheet, css } from 'aphrodite/no-important'

export interface FixedContainerProps {
	title: string
	buttonType: ButtonType
	buttonSize: ButtonSize
	onClick: () => void
	width?: string
	bottom?: number
	top?: number
	left?: number
	right?: number
	marginTop?: number
	marginRight?: number
	marginBottom?: number
	marginLeft?: number
}

export const FixedButton = ({
	title,
	buttonType,
	buttonSize,
	onClick,
	width = '100%',
	bottom,
	top,
	left,
	right,
	marginTop,
	marginRight,
	marginBottom,
	marginLeft,
}: FixedContainerProps) => {
	const styles = StyleSheet.create({
		fixedButtonContainer: {
			position: 'fixed',
			width: '100%',
			bottom,
			top,
			left,
			right,
		},
	})

	return (
		<div className={css(styles.fixedButtonContainer)}>
			<Margin width={'100%'} grow={1} axisDistribution={'center'}>
				<Margin
					width={width}
					direction={'column'}
					marginBottomStep={marginBottom}
					marginTopStep={marginTop}
					marginLeftStep={marginLeft}
					marginRightStep={marginRight}
				>
					<Button title={title} onClick={onClick} buttonType={buttonType} buttonSize={buttonSize} />
				</Margin>
			</Margin>
		</div>
	)
}
