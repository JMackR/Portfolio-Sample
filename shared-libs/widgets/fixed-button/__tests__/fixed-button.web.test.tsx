import React from 'react'
import { render, fireEvent, waitFor } from '@testing-library/react'
import { StyleSheetTestUtils } from 'aphrodite/no-important'
import { createSerializer } from 'jest-aphrodite-react/no-important'
import renderer from 'react-test-renderer'
import { FixedButton, FixedContainerProps } from '../fixed-button'

expect.addSnapshotSerializer(createSerializer(() => StyleSheetTestUtils, { removeVendorPrefixes: true }))

jest.mock('@sample/assets', () => ({
  ActionClear: () => <div>ActionClear</div>,
}))

describe('Fixed Button', () => {
  let props: FixedContainerProps

  describe('has title', () => {
    beforeEach(() => {
      props = {
        onClick: jest.fn(),
        title: 'Test Button',
        buttonType: 'primary',
        buttonSize: 'large',
      }
    })

    it('should render title', () => {
      const { getByText } = render(<FixedButton {...props} />)
      const input = getByText(props.title as string)
      expect(input).toBeDefined()
    })
  })
  describe('is clicked', () => {
    it('should call onClicked', () => {
      const { getByText } = render(<FixedButton {...props} />)
      const button = getByText(props.title as string)
      fireEvent.click(button)
      expect(props.onClick).toHaveBeenCalledTimes(1)
    })
  })
})

describe('Fixed Button Snapshot Tests', () => {
  it('Fixed Button has state true', () => {
    const tree = renderer.create(
      <FixedButton
        onClick={jest.fn()}
        title="Test Button"
        buttonType="primary"
        buttonSize="large"
        bottom={2}
        top={2}
        left={2}
        right={2}
        marginTop={1}
        marginRight={3}
        marginBottom={1}
        marginLeft={3}
      />
    )
    expect(tree).toMatchSnapshot()
  })
})
