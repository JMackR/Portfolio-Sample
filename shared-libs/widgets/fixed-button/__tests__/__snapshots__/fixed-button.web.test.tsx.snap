// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fixed Button Snapshot Tests Fixed Button has state true 1`] = `
.margin {
  margin: undefined;
  margin-left: undefined;
  margin-right: undefined;
  margin-top: undefined;
  margin-bottom: undefined;
  display: flex;
  position: relative;
  flex-basis: auto;
  flex-grow: 1;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: center;
  background-color: undefined;
  height: undefined;
  max-width: undefined;
  width: 100%;
}

.fixedButtonContainer {
  position: fixed;
  width: 100%;
  bottom: 2px;
  top: 2px;
  left: 2px;
  right: 2px;
}

.margin {
  margin: undefined;
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 4px;
  margin-bottom: 4px;
  display: flex;
  position: relative;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: column;
  align-items: stretch;
  justify-content: flex-start;
  background-color: undefined;
  height: undefined;
  max-width: undefined;
  width: 100%;
}

<div
  className="fixedButtonContainer"
>
  <div
    className="margin"
  >
    <div
      className="margin"
    >
      <button
        aria-label=""
        className="sc-bczRLJ cXvykk"
        onClick={[Function]}
      >
        <div
          className="sc-eCYdqJ pmvfX"
        >
          <span
            className="sc-jSMfEi bFVAxJ"
          >
            Test Button
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
